src/main/java/our/yurivongella/instagramclone/service/PostService.java

```diff
@@ -0,0 +1,57 @@
+package our.yurivongella.instagramclone.service;
+
+import com.sun.istack.NotNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import our.yurivongella.instagramclone.controller.dto.post.PostCreateRequestDto;
+import our.yurivongella.instagramclone.controller.dto.post.PostReadResponseDto;
+import our.yurivongella.instagramclone.domain.member.Member;
+import our.yurivongella.instagramclone.domain.member.MemberRepository;
+import our.yurivongella.instagramclone.domain.post.PictureURL;
+import our.yurivongella.instagramclone.domain.post.PictureURLRepository;
+import our.yurivongella.instagramclone.domain.post.Post;
+import our.yurivongella.instagramclone.domain.post.PostRepository;
+import our.yurivongella.instagramclone.util.SecurityUtil;
+
+import java.util.List;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class PostService {
+    final private PostRepository postRepository;
+    final private PictureURLRepository pictureURLRepository;
+    final private MemberRepository memberRepository;
+
+    @Transactional
+    public Long create(PostCreateRequestDto postCreateRequestDto, Long memberId) {
+        Optional<Member> member =memberRepository.findById(SecurityUtil.getCurrentMemberId());
+        Post post = postCreateRequestDto.toPost(member.orElseThrow(() -> new RuntimeException()));
+        post = postRepository.save(post);
+
+        List<PictureURL> list = pictureURLRepository.saveAll(postCreateRequestDto.getPictureURLs(post));
+        post.getPictureURLs().addAll(list);
+
+        return post.getId();
+    }
+
+    public PostReadResponseDto read(Long postId) {
+        Long userId = SecurityUtil.getCurrentMemberId();
+        Member member = memberRepository.findById(userId).get();
+        Optional<Post> post = postRepository.findById(postId);
+
+        return PostReadResponseDto.of(post.orElseThrow(()
+                -> new RuntimeException("게시물이 없습니다.")), member);
+    }
+
+    @Transactional
+    public Long delete(@NotNull Long postId) {
+        Optional<Post> post = postRepository.findById(postId);
+        if(post.orElseThrow(() -> new RuntimeException("게시물이 없습니다.")).getId().equals(SecurityUtil.getCurrentMemberId())) {
+            throw new RuntimeException("잘못된 요청입니다.");
+        }
```

Comment:
위 리뷰와 마찬가지로 `orElseThrow` 를 밖으로 빼는게 좋습니다.
그리고 `post.getId()` 는 **Post 의 ID* 입니다.
`post.getMember().getId()` 를 사용해야 합니다.

```suggestion
        Post post = postRepository.findById(postId)
                .orElseThrow(() -> new RuntimeException("게시물이 없습니다."));

        if (post.getMember().getId().equals(SecurityUtil.getCurrentMemberId())) {
            throw new RuntimeException("게시물 삭제 권한이 없습니다.");
        }
```

Reply:
반영하여 수정하였습니다.
https://github.com/INSPJT/instagram_clone_backend/pull/25/commits/1519df54b1d9b36648e15396918ab756c642a84d